c++/triqs/gfs/legacy_for_python_api.hpp

```diff
@@ -86,5 +86,24 @@ namespace triqs {
     //return r;
     //}
 
+    // basic MeshDLRImFreq operations
+    inline void operator+=(gf_view<dlr_imfreq> g, nda::matrix<std::complex<double>> const &m) {
+      for (int u = 0; u < int(first_dim(g.data())); ++u) g.data()(u, nda::ellipsis()) += m;
+    }
+
+    inline void operator-=(gf_view<dlr_imfreq> g, nda::matrix<std::complex<double>> const &m) {
+      for (int u = 0; u < int(first_dim(g.data())); ++u) g.data()(u, nda::ellipsis()) -= m;
+    }
+
+    inline void operator+=(gf_view<dlr_imfreq> g, std::complex<double> a) { operator+=(g, make_matrix(a * nda::eye<double>(g.target_shape()[0]))); }
+    inline void operator-=(gf_view<dlr_imfreq> g, std::complex<double> a) { operator-=(g, make_matrix(a * nda::eye<double>(g.target_shape()[0]))); }
+
+    // Same for scalar valued
+    // THIS MUST be kept for python operations
+    // specific operations (for legacy python code).
+    inline void operator+=(gf_view<dlr_imfreq, scalar_valued> g, std::complex<double> a) { g.data() += a; }
+
+    inline void operator-=(gf_view<dlr_imfreq, scalar_valued> g, std::complex<double> a) { g.data() -= a; }
+
```

Comment:
The following should cover all the cases implemented above and generalize it for other Mesh types. Please check. Also `make_matrix` may then be redundant.

```suggestion
    // basic MeshDLRImFreq operations
    template <Mesh M, nda::MemoryMatrix Mat>
    inline void operator+=(gf_view<M> g, Mat const &mat) {
      for (auto mp: g.mesh()) g[mp] += mat;
    }

    template <Mesh M, nda::MemoryMatrix Mat>
    inline void operator-=(gf_view<M> g, Mat const &mat) {
      for (auto mp: g.mesh()) g[mp] -= mat;
    }

    template <Mesh M>
    inline void operator+=(gf_view<M> g, dcomplex a) { g += make_regular(a * nda::eye<double>(g.target_shape()[0])); }
    
    template <Mesh M>
    inline void operator-=(gf_view<M> g, dcomplex a) { g -= make_regular(a * nda::eye<double>(g.target_shape()[0])); }

    // Same for scalar valued
    // THIS MUST be kept for python operations
    // specific operations (for legacy python code).
    template <Mesh M>
    inline void operator+=(gf_view<M, scalar_valued> g, dcomplex a) { g.data() += a; }

    template <Mesh M>
    inline void operator-=(gf_view<M, scalar_valued> g, dcomplex a) { g.data() -= a; }

```

Reply:
In the comment here I wrote that this works for MeshDLRImFreq objects, but I guess this is more general now. What includes MemoryMatrix objects? And how can we maybe make in the comment clear what this does? I had a hard time to find this code.